src/utils.tsx

```diff
@@ -157,50 +140,22 @@ export function createCycleSets(types: Type[]): Type[][] {
           ...type.enums.values(),
           ...type.interfaces.values(),
           ...type.namespaces.values(),
-        ];
+        ].filter((t) => shouldReference(program, t));
       default:
         return [];
     }
   }
 
-  // The main recursive function that implements Tarjan's algorithm.
-  function strongConnect(v: Type): void {
-    // Set the depth index for v to the smallest unused index
-    indices.set(v, index);
-    lowlink.set(v, index);
-    index++;
-    stack.push(v);
-    onStack.add(v);
-
-    // Consider successors of v
-    for (const w of referencedTypes(v)) {
-      if (!indices.has(w)) {
-        // Successor w has not yet been visited; recurse on it.
-        strongConnect(w);
-        // After recursion, update lowlink[v]
-        lowlink.set(v, Math.min(lowlink.get(v)!, lowlink.get(w)!));
-      } else if (onStack.has(w)) {
-        // If w is in the current SCC (i.e. on the stack), update lowlink[v]
-        lowlink.set(v, Math.min(lowlink.get(v)!, indices.get(w)!));
+  return {
+    collectType(type: Type) {
+      if (shouldReference(program, type)) {
+        types.add(type);
       }
-    }
-
-    // If v is a root node, pop the stack and generate an SCC.
-    if (lowlink.get(v) === indices.get(v)) {
-      const component: Type[] = [];
-      let w: Type;
-      do {
-        w = stack.pop()!;
-        onStack.delete(w);
-        component.push(w);
-      } while (w !== v);
-
-      const scc = component.filter((v) => inputTypes.has(v));
-      if (scc.length > 0) {
-        sccs.push(scc);
-      }
-    }
-  }
+    },
+    get types() {
+      return [...types.items];
```

Comment:
Whyfor the spread here? Would be a lot more efficient to just return the items set. If `.types` is accessed in a reactive context, every time the set changes it will be re-iterated with a new array allocated.

Reply:
Was making a defensive copy here to not leak set references. Happy to revert.

Reply:
Done!